openapi.yaml

```diff
@@ -3274,7 +3843,67 @@ components:
           $ref: '#/components/schemas/generic-identifier'
         name:
           $ref: '#/components/schemas/generic-identifier'
+
+    temporary-rating:
+      type: object
+      additionalProperties: false
+      description: |
+        Data structure for a temporary rating against a segment.
+        Includes a unique ID, start and (optional) end time, and a reason.  
+
+        May include a set of values, which apply if this temporary rating
+        is either an AAR Exception or temporary seasonal rating.  
+
+      properties:
+        _links:
+          allOf:
+          - $ref: '#/components/schemas/hal-links-block'
```

Comment:
Can we figure out a general strategy to extricate the interop spec from the HAL stuff you need for the LEP UI? Or I need to get over it and prepare to sell HAL to everyone. The interop use cases do not overlap much with the UI use cases, and every vendor is going to have their own UI approach. I have no problem with using relative URL paths as unique identifiers, but I think most of the unique identifiers that are need in interop are specific to the entities doing the exchange.

Reply:
Yeah in all cases (I think) where people have to submit IDs for things, I have included the actual IDs instead of links, because it's awkward.  I'm kind of loathe to remove it now, as I know our customer's TOs are starting ask whether they consider these methods as relatively stable or not.  I haven't heard any pushback on it from other customers so far.  However, to be fair, I also don't think there is a deep understanding of the architectural pattern.  It seems like most users just want to understand what they have to send, and other than that have no comments.  I feel like HAL is useful long-term, although I understand you driving towards minimizing the scope.  

I don't know, just ruminating some thoughts out loud here.  Willing to listen.

Reply:
I fully support HAL in LEP; its use cases will absolutely benefit from a uniform approach to HATEOAS representations. But those aren't TROLIE's interop use cases for the most part. The TROLIE media types are basically as subset, the resource state part. It's worth changing the media type names to reflect that. Your LEP customers can enjoy the uniformity of the HAL envelope and tolerate minor changes to the resource representation schema as both LEP and TROLIE evolve.

Reply:
Removed HAL from TROLIE.